src/node_crypto.cc

```diff
@@ -3563,6 +3563,9 @@ SignBase::Error Sign::SignFinal(const char* key_pem,
                                 const char* passphrase,
                                 unsigned char** sig,
                                 unsigned int *sig_len) {
+  // We have to first clear all stale errors from the shared error stack
+  ERR_clear_error();
```

Comment:
I'm not sure if this is correct.  We normally use `ClearErrorOnReturn clear_error_on_return`, like this:
```diff
diff --git a/src/node_crypto.cc b/src/node_crypto.cc
index 55c4703..2ec2c6b 100644
--- a/src/node_crypto.cc
+++ b/src/node_crypto.cc
@@ -3563,9 +3563,6 @@ SignBase::Error Sign::SignFinal(const char* key_pem,
                                 const char* passphrase,
                                 unsigned char** sig,
                                 unsigned int *sig_len) {
-  // We have to first clear all stale errors from the shared error stack
-  ERR_clear_error();
-
   if (!initialised_)
     return kSignNotInitialised;
 
@@ -3629,6 +3626,9 @@ void Sign::SignFinal(const FunctionCallbackInfo<Value>& args) {
   md_len = 8192;  // Maximum key size is 8192 bits
   md_value = new unsigned char[md_len];
 
+  ClearErrorOnReturn clear_error_on_return;
+  (void) &clear_error_on_return;  // Silence compiler warning.
+
   Error err = sign->SignFinal(
       buf,
       buf_len,
```

But that breaks `test/parallel/test-crypto-rsa-dsa.js` with the following exception:

```
crypto.js:309
  var ret = this._handle.sign(toBuf(key), null, passphrase);
                         ^

Error: error:0906A065:PEM routines:PEM_do_header:bad decrypt
    at Error (native)
    at Sign.sign (crypto.js:309:26)
    at Object.<anonymous> (/home/bnoordhuis/src/v1.x/test/parallel/test-crypto-rsa-dsa.js:140:28)
    at Module._compile (module.js:430:26)
    at Object.Module._extensions..js (module.js:448:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:471:10)
    at startup (node.js:117:18)  
    at node.js:952:3
```

And I can't decide whether that's legitimate or not.